### ISSUE TITLE
TestBackground2D in tests fail due to precision (1.0.1)

### ISSUE THREAD

**Issue by bnavigator** [astropy/photutils#1145]
A few days ago, these tests started to fail while building the package for openSUSE:

```
 =================================== FAILURES ===================================
[  447s] _________________ TestBackground2D.test_resizing[filter_size0] _________________
[  447s] 
[  447s] self = <photutils.background.tests.test_background_2d.TestBackground2D object at 0x7f79ef268340>
[  447s] filter_size = (1, 1)
[  447s] 
[  447s]     @pytest.mark.parametrize('filter_size', FILTER_SIZES)
[  447s]     def test_resizing(self, filter_size):
[  447s]         b1 = Background2D(DATA, (23, 22), filter_size=filter_size,
[  447s]                           bkg_estimator=MeanBackground(), edge_method='crop')
[  447s]         b2 = Background2D(DATA, (23, 22), filter_size=filter_size,
[  447s]                           bkg_estimator=MeanBackground(), edge_method='pad')
[  447s] >       assert_allclose(b1.background, b2.background)
[  447s] E       AssertionError: 
[  447s] E       Not equal to tolerance rtol=1e-07, atol=0
[  447s] E       
[  447s] E       Mismatched elements: 3965 / 10000 (39.6%)
[  447s] E       Max absolute difference: 1.1282854e-06
[  447s] E       Max relative difference: 1.12828667e-06
[  447s] E        x: array([[1., 1., 1., ..., 1., 1., 1.],
[  447s] E              [1., 1., 1., ..., 1., 1., 1.],
[  447s] E              [1., 1., 1., ..., 1., 1., 1.],...
[  447s] E        y: array([[0.999999, 0.999999, 0.999999, ..., 0.999999, 0.999999, 0.999999],
[  447s] E              [0.999999, 0.999999, 0.999999, ..., 0.999999, 0.999999, 0.999999],
[  447s] E              [0.999999, 0.999999, 0.999999, ..., 0.999999, 0.999999, 0.999999],...
[  447s] 
[  447s] ../BUILDROOT/python-photutils-1.0.1-26.1.x86_64/usr/lib64/python3.8/site-packages/photutils/background/tests/test_background_2d.py:85: AssertionError
[  447s] _________________ TestBackground2D.test_resizing[filter_size1] _________________
[  447s] 
[  447s] self = <photutils.background.tests.test_background_2d.TestBackground2D object at 0x7f7a2d64eb80>
[  447s] filter_size = (3, 3)
[  447s] 
[  447s]     @pytest.mark.parametrize('filter_size', FILTER_SIZES)
[  447s]     def test_resizing(self, filter_size):
[  447s]         b1 = Background2D(DATA, (23, 22), filter_size=filter_size,
[  447s]                           bkg_estimator=MeanBackground(), edge_method='crop')
[  447s]         b2 = Background2D(DATA, (23, 22), filter_size=filter_size,
[  447s]                           bkg_estimator=MeanBackground(), edge_method='pad')
[  447s] >       assert_allclose(b1.background, b2.background)
[  447s] E       AssertionError: 
[  447s] E       Not equal to tolerance rtol=1e-07, atol=0
[  447s] E       
[  447s] E       Mismatched elements: 3965 / 10000 (39.6%)
[  447s] E       Max absolute difference: 1.1282854e-06
[  447s] E       Max relative difference: 1.12828667e-06
[  447s] E        x: array([[1., 1., 1., ..., 1., 1., 1.],
[  447s] E              [1., 1., 1., ..., 1., 1., 1.],
[  447s] E              [1., 1., 1., ..., 1., 1., 1.],...
[  447s] E        y: array([[0.999999, 0.999999, 0.999999, ..., 0.999999, 0.999999, 0.999999],
[  447s] E              [0.999999, 0.999999, 0.999999, ..., 0.999999, 0.999999, 0.999999],
[  447s] E              [0.999999, 0.999999, 0.999999, ..., 0.999999, 0.999999, 0.999999],...
[  447s] 
[  447s] ../BUILDROOT/python-photutils-1.0.1-26.1.x86_64/usr/lib64/python3.8/site-packages/photutils/background/tests/test_background_2d.py:85: AssertionError
[  447s] _______________ TestBackground2D.test_background_mask[box_size0] _______________
[  447s] 
[  447s] self = <photutils.background.tests.test_background_2d.TestBackground2D object at 0x7f7a2d64cee0>
[  447s] box_size = (25, 25)
[  447s] 
[  447s]     @pytest.mark.parametrize('box_size', ([(25, 25), (23, 22)]))
[  447s]     def test_background_mask(self, box_size):
[  447s]         """
[  447s]         Test with an input mask.  Note that box_size=(23, 22) tests the
[  447s]         resizing of the image and mask.
[  447s]         """
[  447s]     
[  447s]         data = np.copy(DATA)
[  447s]         data[25:50, 25:50] = 100.
[  447s]         mask = np.zeros(DATA.shape, dtype=bool)
[  447s]         mask[25:50, 25:50] = True
[  447s]         b = Background2D(data, box_size, filter_size=(1, 1), mask=mask,
[  447s]                          bkg_estimator=MeanBackground())
[  447s] >       assert_allclose(b.background, DATA)
[  447s] E       AssertionError: 
[  447s] E       Not equal to tolerance rtol=1e-07, atol=0
[  447s] E       
[  447s] E       Mismatched elements: 9419 / 10000 (94.2%)
[  447s] E       Max absolute difference: 1.39388819e-05
[  447s] E       Max relative difference: 1.39388819e-05
[  447s] E        x: array([[0.999986, 0.999986, 0.999986, ..., 0.999993, 0.999993, 0.999993],
[  447s] E              [0.999986, 0.999986, 0.999987, ..., 0.999993, 0.999993, 0.999993],
[  447s] E              [0.999986, 0.999987, 0.999987, ..., 0.999993, 0.999993, 0.999993],...
[  447s] E        y: array([[1., 1., 1., ..., 1., 1., 1.],
[  447s] E              [1., 1., 1., ..., 1., 1., 1.],
[  447s] E              [1., 1., 1., ..., 1., 1., 1.],...
[  447s] 
[  447s] ../BUILDROOT/python-photutils-1.0.1-26.1.x86_64/usr/lib64/python3.8/site-packages/photutils/background/tests/test_background_2d.py:101: AssertionError
[  447s] _______________ TestBackground2D.test_background_mask[box_size1] _______________
[  447s] 
[  447s] self = <photutils.background.tests.test_background_2d.TestBackground2D object at 0x7f7a2d588460>
[  447s] box_size = (23, 22)
[  447s] 
[  447s]     @pytest.mark.parametrize('box_size', ([(25, 25), (23, 22)]))
[  447s]     def test_background_mask(self, box_size):
[  447s]         """
[  447s]         Test with an input mask.  Note that box_size=(23, 22) tests the
[  447s]         resizing of the image and mask.
[  447s]         """
[  447s]     
[  447s]         data = np.copy(DATA)
[  447s]         data[25:50, 25:50] = 100.
[  447s]         mask = np.zeros(DATA.shape, dtype=bool)
[  447s]         mask[25:50, 25:50] = True
[  447s]         b = Background2D(data, box_size, filter_size=(1, 1), mask=mask,
[  447s]                          bkg_estimator=MeanBackground())
[  447s] >       assert_allclose(b.background, DATA)
[  447s] E       AssertionError: 
[  447s] E       Not equal to tolerance rtol=1e-07, atol=0
[  447s] E       
[  447s] E       Mismatched elements: 3965 / 10000 (39.6%)
[  447s] E       Max absolute difference: 1.1282854e-06
[  447s] E       Max relative difference: 1.1282854e-06
[  447s] E        x: array([[0.999999, 0.999999, 0.999999, ..., 0.999999, 0.999999, 0.999999],
[  447s] E              [0.999999, 0.999999, 0.999999, ..., 0.999999, 0.999999, 0.999999],
[  447s] E              [0.999999, 0.999999, 0.999999, ..., 0.999999, 0.999999, 0.999999],...
[  447s] E        y: array([[1., 1., 1., ..., 1., 1., 1.],
[  447s] E              [1., 1., 1., ..., 1., 1., 1.],
[  447s] E              [1., 1., 1., ..., 1., 1., 1.],...
[  447s] 
[  447s] ../BUILDROOT/python-photutils-1.0.1-26.1.x86_64/usr/lib64/python3.8/site-packages/photutils/background/tests/test_background_2d.py:101: AssertionError
[  447s] ______________________ TestBackground2D.test_zero_padding ______________________
[  447s] 
[  447s] self = <photutils.background.tests.test_background_2d.TestBackground2D object at 0x7f7a2d577430>
[  447s] 
[  447s]     def test_zero_padding(self):
[  447s]         """Test case where padding is added only on one axis."""
[  447s]     
[  447s]         b = Background2D(DATA, (25, 22), filter_size=(1, 1))
[  447s] >       assert_allclose(b.background, DATA)
[  447s] E       AssertionError: 
[  447s] E       Not equal to tolerance rtol=1e-07, atol=0
[  447s] E       
[  447s] E       Mismatched elements: 8548 / 10000 (85.5%)
[  447s] E       Max absolute difference: 7.53360419e-06
[  447s] E       Max relative difference: 7.53360419e-06
[  447s] E        x: array([[0.999992, 0.999992, 0.999993, ..., 0.999993, 0.999993, 0.999993],
[  447s] E              [0.999993, 0.999993, 0.999993, ..., 0.999993, 0.999993, 0.999993],
[  447s] E              [0.999993, 0.999993, 0.999993, ..., 0.999993, 0.999993, 0.999993],...
[  447s] E        y: array([[1., 1., 1., ..., 1., 1., 1.],
[  447s] E              [1., 1., 1., ..., 1., 1., 1.],
[  447s] E              [1., 1., 1., ..., 1., 1., 1.],...
[  447s] 
[  447s] ../BUILDROOT/python-photutils-1.0.1-26.1.x86_64/usr/lib64/python3.8/site-packages/photutils/background/tests/test_background_2d.py:159: AssertionError
```

Full build log with all package versions: 
[photutils-1.0.1_fail_log.txt](https://github.com/astropy/photutils/files/5820123/photutils-1.0.1_fail_log.txt)

Do I have to be concerned that some updated library produces less precision as it should, or can I simply ignore these tests? It might be due to scipy 1.6.0, which was updated in the openSUSE build system the same day as the [photutils builds started to fail](https://build.opensuse.org/packages/python-photutils/job_history/devel:languages:python:numeric/openSUSE_Tumbleweed/x86_64).

**Comment by larrybradley** [comment]
Yes, this was due to upstream changes in `scipy 1.6.0` `ndimage.zoom` boundary handling (https://github.com/astropy/photutils/issues/1136).  I put in a change in https://github.com/astropy/photutils/pull/1137 that gets back the old results (to be released in v1.0.2, likely next week).  The scipy 1.6.0 differences are very small and they only affect the array edges (the test arrays are small, so all pixels are affected).  But please let me know if you happen to see any big changes.

**Comment by bnavigator** [comment]
Thank you, I should have searched for scipy 1.6.0 in the issues.
